### PR TITLE
API improvements for custom column generation

### PR DESCRIPTION
**Pull request by tobiasqueck**
@stefanschreck, here are the discussed changes:
- removed `template` property but allowed defining the target path via `folder` property
- removed `eventHandler` object from the method signature and added boolean for handler and option to provide custom xml
- create `CustomElement` type that is to be used as base for FPM elements that this module allows to generate
- enhanced the the `CustomPage` and `CustomAction` API to be aligned with `CustomColumn`

### CODE REVIEW

**Comment by tobiasqueck** [comment]
@zdravko-georgiev please have a look at the changes to the `CustomPage` API. The changes are based on your comment https://github.com/SAP/open-ux-tools/pull/39#issuecomment-954883159 and a discussion with @stefanschreck about the `CustomColumn` API.
This PR has the goal to consolidate both APIs as well as providing more flexibility for consumers.

**Review by stefanschreck — APPROVED** [review]
Much better now.
@zdravko-georgiev Could you please retest the consumption based on this branch?

**Comment by zdravko-georgiev** [comment]
> @zdravko-georgiev please have a look at the changes to the `CustomPage` API. The changes are based on your comment [#39 (comment)](https://github.com/SAP/open-ux-tools/pull/39#issuecomment-954883159) and a discussion with @stefanschreck about the `CustomColumn` API. This PR has the goal to consolidate both APIs as well as providing more flexibility for consumers.

@tobiasqueck thanks, I will review and retest now.

File: packages/fev4-fpm-writer/src/column/version.ts [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
     validateVersion(ui5Version);
     if (ui5Version === undefined || ui5Version >= 1.86) {
-        return join(__dirname, '..', '..', 'templates', 'column', '1.86');
+        return join(__dirname, '../../templates/column/1.86');
     } else if (ui5Version === 1.85) {
-        return join(__dirname, '..', '..', 'templates', 'column', '1.85');
+        return join(__dirname, '../../templates/column/1.85');
     } else {
-        return join(__dirname, '..', '..', 'templates', 'column', '1.84');
+        return join(__dirname, '../../templates/column/1.84');
```
Comment (zdravko-georgiev) [inline packages/fev4-fpm-writer/src/column/version.ts:17]: Is this going to work on windows? I am asking because of the path with forward slashes

Comment (stefanschreck) [reply packages/fev4-fpm-writer/src/column/version.ts:17]: @zdravko-georgiev join does this automatically, said Tobias

Comment (tobiasqueck) [reply packages/fev4-fpm-writer/src/column/version.ts:17]: https://nodejs.org/api/path.html#pathjoinpaths

**Review by zdravko-georgiev — APPROVED** [review]
Tested successfully

**Pull request merged**